projects/plugins/jetpack/extensions/blocks/dialogue/edit.js

```diff
@@ -93,16 +93,18 @@ export default function DialogueEdit( {
 
 	// Update dialogue participant with conversation participant changes.
 	useEffect( () => {
-		if ( ! conversationParticipant ) {
+		// Do not update current Dialogue block.
+		if ( isSelected ) {
 			return;
 		}
 
-		if ( conversationParticipant.slug !== slug ) {
+		// When no context, nothing to do.
+		if ( ! conversationParticipant ) {
 			return;
 		}
 
-		// Do not update current Dialogue block.
-		if ( isSelected ) {
+		// Only take care of Dialogue with same speaker.
+		if ( conversationParticipant.slug !== slug ) {
```

Comment:
It would be worth checking that the `conversationParticipant` object reference isn't changing when the `slug` and `label` properties aren't. I've seen this happen and had to list the specific properties being used in the dependency array `[ conversationParticipant.label, ...` etc.

Reply:
👍 Let me do it in a follow-up PR. I don't wanna lose the ✅